### PR TITLE
gc_worker & service/kv: Implement Non-streaming physical_scan_lock

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

###  What have you changed?

This PR changes `physical_scan_lock` to non-streaming implementation. This is used to solve the issue that TiDB cannot block TiKV's sender when TiDB's resolving speed of locks is slower than TiKV's sending speed, which may cause TiDB consumes too much memory to buffer the locks.

The old streaming physical_scan_lock is directly changed to the streaming one. Since the feature is not enabled in any revision, so it's safe not to care about the compatibility.

###  What is the type of the changes?

- Bugfix (a change which fixes an issue)

###  How is the PR tested?

- Unit test
- Integration test
- Manual test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### Related changes

- [x] kvproto: https://github.com/pingcap/kvproto/pull/553

### CODE REVIEW

**Comment by MyonKeminta** [comment]
## Test Result
* The memory issue is fixed.
* A schrodinger case is now running to check whether this change introduces incorrectness.

Before:
image
After:
image

**Comment by sticnarf** [comment]
By the way, L1226 of `gc_worker.rs` should be `ts <= 11`?

File: src/server/gc_worker/gc_worker.rs [hunk 1]
```diff
@@ -577,43 +486,53 @@ impl<E: Engine> GcRunner<E> {
...
+        let mut lock_infos = Vec::with_capacity(locks.len());
+        for (key, lock) in locks {
+            let raw_key = key.into_raw().map_err(MvccError::from)?;
+            //     Ok(k) => k,
+            //     Err(e) => {
+            //         self.is_finished = true;
+            //         let e = Error::from(MvccError::from(e));
+            //         return Err(e);
+            //     }
+            // };
```
[the remark this answers is not in the document]
Comment (MyonKeminta) [reply src/server/gc_worker/gc_worker.rs]: Oh, sorry, I forgot

**Comment by MyonKeminta** [comment]
???
/test

**Comment by sticnarf** [comment]
> LGTM. BTW, do we need to support batch `physical_scan_lock`? https://github.com/pingcap/kvproto/blob/master/proto/tikvpb.proto#L76

I think it should. It seems the corresponding TiDB PR also does not add it to the batch commands list in https://github.com/pingcap/tidb/blob/master/store/tikv/tikvrpc/tikvrpc.go#L358

**Comment by MyonKeminta** [comment]
> LGTM. BTW, do we need to support batch `physical_scan_lock`? https://github.com/pingcap/kvproto/blob/master/proto/tikvpb.proto#L76

Oh, yes, we should. I forgot it. I think I'll add it in later PRs.

**Pull request merged**